simulatingrisk/hawkdove/model.py

```diff
@@ -0,0 +1,176 @@
+from enum import Enum
+import math
+
+import mesa
+
+from simulatingrisk.utils import coinflip
+
+Play = Enum("Play", ["HAWK", "DOVE"])
+play_choices = [Play.HAWK, Play.DOVE]
+
+
+# divergent color scheme, nine colors
+# from https://colorbrewer2.org/?type=diverging&scheme=RdYlGn&n=9
+divergent_colors_9 = [
+    "#d73027",
+    "#f46d43",
+    "#fdae61",
+    "#fee08b",
+    "#ffffbf",
+    "#d9ef8b",
+    "#a6d96a",
+    "#66bd63",
+    "#1a9850",
+]
+
+# divergent color scheme, ficolors
+# from https://colorbrewer2.org/?type=diverging&scheme=RdYlGn&n=5
+divergent_colors_5 = ["#d7191c", "#fdae61", "#ffffbf", "#a6d96a", "#1a9641"]
+
+
+class HawkDoveAgent(mesa.Agent):
+    """
+    An agent with a risk attitude playing Hawk or Dove
+    """
+
+    def __init__(self, unique_id, model, risk_level=None, hawk_odds=None):
+        super().__init__(unique_id, model)
+
+        self.points = 0
+        self.choice = self.initial_choice(hawk_odds)
+        self.last_choice = None
+
+        # risk level
+        # - based partially on neighborhood size,
+        #  which is configurable at the model level
+        num_neighbors = 8 if self.model.include_diagonals else 4
+        self.risk_level = risk_level or self.random.randint(0, num_neighbors)
+
+    def initial_choice(self, hawk_odds=None):
+        # first round : choose what to play randomly or based on initial hawk odds
+        opts = {}
+        if hawk_odds:
+            opts["weight"] = hawk_odds
+        return coinflip(play_choices, **opts)
+
+    @property
+    def neighbors(self):
+        # use configured neighborhood (with or without diagonals) on the model;
+        # don't include the current agent
+        return self.model.grid.get_neighbors(
+            self.pos, moore=self.model.include_diagonals, include_center=False
+        )
+
+    @property
+    def num_dove_neighbors(self):
+        """count how many neighbors played DOVE on the last round"""
+        return len([n for n in self.neighbors if n.last_choice == Play.DOVE])
+
+    def choose(self):
+        "decide what to play this round"
+        # after the first round, choose based on what neighbors did last time
+        if self.model.schedule.steps > 0:
+            # store previous  choice
+            self.last_choice = self.choice
+
+            # choose based on the number of neighbors who played
+            # dove last round and agent risk level
+
+            # agent with r = 0 should always take the risky choice
+            #   (any risk is acceptable).
+            # agent with r = max should always take the safe option
+            #   (no risk is acceptable)
+            if self.num_dove_neighbors > self.risk_level:
+                self.choice = Play.HAWK
+            else:
+                self.choice = Play.DOVE
```

Comment:
@LaraBuchak here's the revised logic for hawk/dove agent choice of what to play, based on your revision to the logic. Determining based on number of neighbors who played dove.